src/Bundle/Doctrine/DBAL/DataSource.php

```diff
@@ -62,7 +64,7 @@ public function getData(Parameters $parameters)
 
         $paginator = new Pagerfanta(new QueryAdapter($this->queryBuilder, $countQueryBuilderModifier));
         $paginator->setNormalizeOutOfRangePages(true);
-        $paginator->setCurrentPage($parameters->get('page', 1));
+        $paginator->setCurrentPage($page > 0 ? $page : 1);
```

Comment:
Did we wanted to cover some specific case with this comparison? We didn't cover it with tests

Reply:
Yes I can add a test with that behaviour. This has been added to fix a Psalm error.

Reply:
If so, I'm fine with it, but still would be better to cover with additional test I suppose

Reply:
Just for reference, the comparison here does two things:

1) Prevents Pagerfanta's `LessThan1CurrentPageException` from bubbling up if someone passes a negative number or 0 through the page parameter
2) Pagerfanta 3.x is a lot more strictly typed, both in terms of PHP and static analysis tools (I haven't figured out Psalm yet, but PHPStan is used pretty heavily on my repos, and both of those tools parse each other's doc block annotations which is a good thing).  This specific method has a `@phpstan-param positive-int` for the page parameter, so SA is doing the right thing by flagging it if it can't guarantee it's a positive integer, and the exception at runtime for invalid input is still there for less strictly typed code.